### ISSUE TITLE
We should decide how we are supposed to display duplicate website visits in the HistoryPanel

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#23515]
After fixing [https://mozilla-hub.atlassian.net/browse/FXIOS-10740](https://mozilla-hub.atlassian.net/browse/FXIOS-10740|smart-link) , there is a small change in behaviour with the HistoryPanel.

For comparison, on Desktop, a website visited several times in a row (with no other websites in between) only appears once in history. We are no longer aggregating repeated views of the same website.

**Example**:

!Screenshot 2024-12-02 at 12.20.11 PM.jpeg|width=33.33333333333333%,alt="Screenshot 2024-12-02 at 12.20.11 PM.jpeg"!

Basically, this was probably only working for us before by accident rather than explicit intention.

Development-wise, we need to consider more than simply filtering the URL. Chances are we expect the same site to appear multiple times in history if the user visits it today, yesterday, or a month ago. However, we need some clarification from product what the actual exact behaviour should be for this screen.

┆Issue is synchronized with this [Jira Spike](https://mozilla-hub.atlassian.net/browse/FXIOS-10779)

**Comment by data-sync-user** [comment]
➤ Isabella commented:

Andres Furlan We need to check with what the expected behaviour should be here.

**Comment by data-sync-user** [comment]
➤ Isabella commented:

FYI this is needed to address the following regression bug: https://mozilla-hub.atlassian.net/browse/FXIOS-10808 ( https://mozilla-hub.atlassian.net/browse/FXIOS-10808|smart-link )

**Comment by data-sync-user** [comment]
➤ Nishant Bhasin commented:

Andres Furlan Could you look into this please!!

**Comment by data-sync-user** [comment]
➤ Isabella commented:

Related bug tickets:

https://mozilla-hub.atlassian.net/browse/FXIOS-10808 ( https://mozilla-hub.atlassian.net/browse/FXIOS-10808|smart-link ) 

https://mozilla-hub.atlassian.net/browse/FXIOS-10868 ( https://mozilla-hub.atlassian.net/browse/FXIOS-10868|smart-link )  (closed duplicate)

**Comment by data-sync-user** [comment]
➤ Isabella commented:

cc Andres Furlan Several people on our team have asked about this one… maybe we can get some movement on what the expected desktop behaviour is so someone can tackle this? 

I suspect this could be anywhere between 2-6 days of work…

**Comment by data-sync-user** [comment]
➤ Isabella commented:

See implementation details: https://mozilla-hub.atlassian.net/browse/FXIOS-10808?focusedCommentId=1011121 ( https://mozilla-hub.atlassian.net/browse/FXIOS-10808?focusedCommentId=1011121|smart-link )

**Comment by data-sync-user** [comment]
➤ Alina Moldovan commented:

Isabella Yoana Rios Diaz Could you please add a few details so QA knows what to test? It is not clear for me what is the expected behavior now, except the fact that the sites would not be duplicated in the history list.

**Comment by data-sync-user** [comment]
➤ Isabella commented:

Hi Alina Moldovan , when you visit a single website, you should only see 1 item appear in history now. Previously (as in the ticket screenshot), you could visit a website or do a search and you’d see multiple repeated / duplicate entries in the history panel. You should only see a duplicate entry in history if you actually navigated to that website again later after visiting another website, for example. I hope that is clear enough, let me know if you need us to provide additional clarification! 🙏

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Isabella Thank you! Please note that I’ve tried on v143.2 (62218) with iPhone 13 Pro (iOS 26). There are still a lot of duplicate entries in history. 

!image-20251002-134105.png|width=359,alt="image-20251002-134105.png"!

**Comment by data-sync-user** [comment]
➤ Isabella commented:

Thanks for checkingAndrei Bodea ! 🙏  It looks to me that the actual URLs are different here (in the subtitle of each row, at the very end… after the ? for example).

Can you describe a bit how you were browsing to create these? If you were clicking ads or results and then navigating back to the page I think I would expect to see some extra Google entries, for example.

Also, a short demo video might be helpful for me and Yoana Rios Diaz to confirm based on your web browsing, if those entries make sense. 🙏

**Comment by data-sync-user** [comment]
➤ Yoana Rios Diaz commented:

Hi Andrei Bodea related to search visit I did notice there are several records per visit depending on the amount of redirections the search does but we have this issue before when Search Terms groups were enabled. The fix focused on the individual visits getting recorded twice. 

We could have a follow up ticket to decided what to do with this cases

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Validated on v144(62506), with iPhone 16+(26).

[^ScreenRecording_10-12-2025 16-05-45_1.MP4]
